linera-views/src/common.rs

```diff
@@ -41,6 +46,26 @@ pub(crate) enum Update<T> {
 /// The minimum value for the view tags. Values in 0..MIN_VIEW_TAG are used for other purposes.
 pub(crate) const MIN_VIEW_TAG: u8 = 1;
 
+/// Data type indicating that the database is not consistent
+#[derive(Error, Debug)]
+pub enum DatabaseConsistencyError {
+    /// The key is of length less than 4 so we cannot extract the first byte
+    #[error("key of length less than 4, so we cannot extract the first byte")]
+    TooShortKey,
+
+    /// In the splitting scheme a certain number of segments was specified, but some are missing
+    #[error("segment is missing from the database")]
+    MissingSegment,
+
+    /// We expect that the segments arrive sequentially
+    #[error("segment index is not what was expected")]
+    IncorrectIndexEntry,
+
+    /// In the value, we should have a count entry.
```

Comment:
In general, comments on error variants are useless because an error variant always has an explanation string that should be clear already. If we don't want to use `#[clippy(allow:missing_docs)]`, I would just use the same sentence as the error string.

Reply:
Ok, I use systematically the error string as the documentation. I find `missing_docs` something to avoid usually.